include/alpaka/dev/cpu/SysInfo.hpp

```diff
@@ -59,7 +59,7 @@ namespace alpaka
             namespace detail
             {
 #if BOOST_ARCH_X86
-    #if BOOST_COMP_GNUC || BOOST_COMP_CLANG || (!BOOST_COMP_MSVC_EMULATED && __INTEL_COMPILER)
+    #if BOOST_COMP_GNUC || BOOST_COMP_CLANG || (!BOOST_COMP_MSVC_EMULATED && defined(__INTEL_COMPILER))
```

Comment:
@BenjaminW3 
- Is this `defined` fix only an issue with CUDA 9.2 or is it an general issue? 
- Do we need this `defined` to have INTEL support or is is only a stile change?